packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -7,22 +7,59 @@
 import {ApiClients, Argument, DataType} from '../types'
 import {useAsync} from '../useAsync'
 import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
-
 /**
  * A hook for `ShopperProducts#getProducts`.
  * Allows access to multiple products by a single request. Only products that are online and assigned to a site catalog are returned. The maximum number of productIDs that can be requested are 24. Along with product details, the availability, images, price, promotions, and variations for the valid products will be included, as appropriate.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=getProducts} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getproducts} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProducts = (
-    arg: Argument<Client['getProducts']>
-): UseQueryResult<DataType<Client['getProducts']>, Error> => {
+type UseProductsParameters = NonNullable<Argument<Client['getProducts']>>['parameters']
+type UseProductsHeaders = NonNullable<Argument<Client['getProducts']>>['headers']
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: false
+        queryOptions: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProducts']>, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: true
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<Response, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: boolean
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+): UseQueryResult<DataType<Client['getProducts']> | Response, Error>
+function useProducts(
+    parameters: UseProductsParameters,
+    options?: {
+        headers: UseProductsHeaders
+        rawResponse: boolean
+        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+    }
+) {
```

Comment:
Looks like the last signature is a duplicate. I've tried deleting it, and Typescript compiler did not give me any error.

```suggestion
function useProducts(
    parameters: UseProductsParameters,
    options?: {
        headers: UseProductsHeaders
        rawResponse: boolean
        queryOptions?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
    }
): UseQueryResult<DataType<Client['getProducts']> | Response, Error> {
```

Reply:
For overloads, the function implementation does not mean function signature, we need to add a case of `boolean` on top of the implementation, it looks like a duplication,but it is not

Reply:
Both signatures above have `rawResponse: boolean`. Yup I see why there needs to be signature that has that `boolean`. But there shouldn't be 2 signatures like that, right?

Currently, there are 4 total signatures, but I think it should be 3:
1. With `rawResponse: true` ← overloading signature
1. With `rawResponse: false` ← overloading signature
1. With `rawResponse: boolean` ← implementation signature